### ISSUE TITLE
The snap version of meshlab is outdated

### ISSUE THREAD

**Issue by lianghongzhuo** [cnr-isti-vclab/meshlab#691]
#### Check all platforms you experienced the issue (change to `[x]`)
- [ ] Windows
- [ ] Every platform
- [X] Linux
- [ ] Mac OS

#### What type of issue is?
- [ ] bug or unexpected behavior
- [ ] crash
- [X] feature request
- [ ] other (specify...)

#### Describe your issue

The snap version of meshlab is outdated. https://snapcraft.io/meshlab
Is there any plan to push the latest binary to the snap store?

Thank you.

**Comment by alemuntoni** [comment]
Hi, it's not a priority at the moment, since we are releasing appimages for linux.
But pull requests for the generation of a snap package are welcome.

**Comment by lianghongzhuo** [comment]
I tried to build a snap using `snapcraft`
I got this error:
```
make[1]: Entering directory '/root/parts/meshlab/build/meshlabplugins/edit_pickpoints'
make[1]: Nothing to be done for 'first'.
make[1]: Leaving directory '/root/parts/meshlab/build/meshlabplugins/edit_pickpoints'
cp -f -R "/root/parts/meshlab/build/src/../distrib" "/root/parts/meshlab/build"
cp: '/root/parts/meshlab/build/src/../distrib' and '/root/parts/meshlab/build/distrib' are the same file
Makefile:1992: recipe for target 'copydir' failed
make: *** [copydir] Error 1
```
It seems that the `copydir` command wants to copy the files with the same orientation and destination.
Any suggestions to fix it?
I'm using ubuntu18, and this is the modified `snapcraf.yaml`

```
name: meshlab
base: core18
version: '2020.05'
summary: MeshLab
description: |
  The open source system for processing and editing 3D triangular meshes.
  It provides a set of tools for editing, cleaning, healing, inspecting,
  rendering, texturing and converting meshes. It offers features for processing
  raw data produced by 3D digitization tools/devices and for preparing models
  for 3D printing.
icon: snap/gui/meshlab.png

grade: stable
confinement: strict

apps:
  meshlab:
    command: desktop-launch meshlab
    plugs: [home, x11, mir, opengl]
  meshlabserver:
    command: desktop-launch meshlabserver
    plugs: [home, x11, opengl]


parts:
  meshlab:
    # after: [desktop-qt5]
    plugin: qmake
    qt-version: qt5
    source: https://github.com/cnr-isti-vclab/meshlab.git
    build-packages:
      - qt5-default
      - qttools5-dev-tools
      - qtscript5-dev
      - libqt5xmlpatterns5-dev
      - mesa-common-dev
      - libglu1-mesa-dev
      - lib3ds-dev
      - libglew-dev
      - libeigen3-dev 
      - libopenctm-dev 
      - libgmp-dev 
      - libqhull-dev
    override-build: |
      qmake src/meshlab.pro "CONFIG+= system_eigen3 system_glew system_bzip2 system_openctm system_lib3ds"
      make -j4
```

**Comment by alemuntoni** [comment]
It is possible to change the `override-build` to:
```
cd src
qmake meshlab.pro "CONFIG+= system_eigen3 system_glew system_bzip2 system_openctm system_lib3ds"
make -j4
```
There is a bug on the qmake config when qmake is run from the main directory of meshlab, I'll fix asap.

**Comment by lianghongzhuo** [comment]
Yes, I already changed like that. I'm still testing the snap process. I hope no more error pumps out. I will report the result later.

**Comment by alemuntoni** [comment]
Cool, thanks!
In the meantime, I fixed the bug on the qmake build and `qmake src/meshlab.pro` should now work.

**Comment by lianghongzhuo** [comment]
Hi, I raised a pull request to update the snap.
You can use `SNAPCRAFT_BUILD_ENVIRONMENT_CPU=6 SNAPCRAFT_BUILD_ENVIRONMENT_MEMORY=8G snapcraft` to make the snap file. This can overwrite the default multipass virtual machine settings.

**Comment by lianghongzhuo** [comment]
The second comment should also fix #367

**Comment by alemuntoni** [comment]
Great, thank you!! 
I'll set up a github workflow for automatically update the version and build the package, and then we will upload the new version on the snap store (probably will be 2020.06)!

**Comment by lianghongzhuo** [comment]
Thank you!

**Comment by lianghongzhuo** [comment]
Hi, I tested the auto-build snap from https://github.com/cnr-isti-vclab/meshlab/suites/720118171/artifacts/7125140
However, when I open meshlab, I got such error: `meshlab: error while loading shared libraries: libmeshlab-common.so.1: cannot open shared object file: No such file or directory
`
The snap file build on my local computer (ubuntu 18) works well.  I don't know what happened.
Debian community also has a similar bug month ago, they fixed it by https://salsa.debian.org/science-team/meshlab/-/commit/8bf6ba90d7197143ec3211d8470a85475d30d7fa But I don't know whether it is relevant?

**Comment by alemuntoni** [comment]
I am trying to make something similar to the [appimage structure](https://github.com/cnr-isti-vclab/meshlab/blob/50dc7d421ff835bc77a171253c378251f2881da7/install/linux/linux_appimages.sh#L47) inside the snap.

**Comment by alemuntoni** [comment]
@lianghongzhuo could you please try the snap generated in [this](https://github.com/alemuntoni/meshlab/actions/runs/116694897) workflow run, and tell me if it works?
Works on ubuntu 20.04.

**Comment by lianghongzhuo** [comment]
This snap also works in ubuntu 18.04, thanks!

**Comment by alemuntoni** [comment]
Ok great, merged! Thanks @lianghongzhuo!
